### PR TITLE
Update README and JavaDoc for passkeys

### PR DESCRIPTION
**Pull request by emlun**

### CODE REVIEW

File: README [hunk 1]
```diff
@@ -367,31 +370,56 @@ throw new RuntimeException("Authentication failed");
...
+
+A https://passkeys.dev/[passkey] is a WebAuthn credential that can simultaneously both _identify_ and _authenticate_ the user.
+This is also called a link:https://www.w3.org/TR/2021/REC-webauthn-2-20210408/#discoverable-credential[discoverable credential].
+By default, credentials are created non-discoverable, which means the server
+must list them in the
+https://www.w3.org/TR/webauthn/#dom-publickeycredentialrequestoptions-allowcredentials[`allowCredentials`]
+parameter before the user can use them to authenticate.
+This is typically because the credential private key is not stored within the authenticator,
+but instead encoded into one of the credential IDs in `allowCredentials`.
+This way even a small hardware authenticator can have an unlimited credential capacity,
```
Comment (aveenismail) [inline README:411]: I would put comma after "This way" and remove it from after "capacity". If you think the sentence would become too long, I would structure it to something like: ... credential capacity. However, there is still the drawback that...

File: webauthn-server-core/src/main/java/com/yubico/webauthn/AssertionRequest.java [hunk 2]
```diff
@@ -59,8 +59,10 @@ public class AssertionRequest {
    * <p>If both this and {@link #getUserHandle() userHandle} are empty, this indicates that this is
    * a request for an assertion by a <a
    * href="https://www.w3.org/TR/2021/REC-webauthn-2-20210408/#client-side-discoverable-public-key-credential-source">client-side-discoverable
-   * credential</a>, and identification of the user has been deferred until the response is
-   * received.
+   * credential</a> (passkey), and identification of the user has been deferred until the response
```
Comment (aveenismail) [inline webauthn-server-core/src/main/java/com/yubico/webauthn/AssertionRequest.java:62]: I would remove the comma before "and"

File: webauthn-server-core/src/main/java/com/yubico/webauthn/AssertionRequest.java [hunk 3]
```diff
@@ -74,8 +76,10 @@ public class AssertionRequest {
    * <p>If both this and {@link #getUsername() username} are empty, this indicates that this is a
    * request for an assertion by a <a
    * href="https://www.w3.org/TR/2021/REC-webauthn-2-20210408/#client-side-discoverable-public-key-credential-source">client-side-discoverable
-   * credential</a>, and identification of the user has been deferred until the response is
-   * received.
+   * credential</a> (passkey), and identification of the user has been deferred until the response
```
Comment (aveenismail) [inline webauthn-server-core/src/main/java/com/yubico/webauthn/AssertionRequest.java:79]: I would remove the comma before "and"

File: webauthn-server-core/src/main/java/com/yubico/webauthn/AssertionRequest.java [hunk 4]
```diff
@@ -105,8 +109,10 @@ private AssertionRequest(
    * <p>If both this and {@link #getUserHandle()} are empty, this indicates that this is a request
    * for an assertion by a <a
    * href="https://www.w3.org/TR/2021/REC-webauthn-2-20210408/#client-side-discoverable-public-key-credential-source">client-side-discoverable
-   * credential</a>, and identification of the user has been deferred until the response is
-   * received.
+   * credential</a> (passkey), and identification of the user has been deferred until the response
```
Comment (aveenismail) [inline webauthn-server-core/src/main/java/com/yubico/webauthn/AssertionRequest.java:112]: I would remove the comma before "and"

File: webauthn-server-core/src/main/java/com/yubico/webauthn/AssertionRequest.java [hunk 5]
```diff
@@ -121,8 +127,10 @@ public Optional<String> getUsername() {
    * <p>If both this and {@link #getUsername()} are empty, this indicates that this is a request for
    * an assertion by a <a
    * href="https://www.w3.org/TR/2021/REC-webauthn-2-20210408/#client-side-discoverable-public-key-credential-source">client-side-discoverable
-   * credential</a>, and identification of the user has been deferred until the response is
-   * received.
+   * credential</a> (passkey), and identification of the user has been deferred until the response
```
Comment (aveenismail) [inline webauthn-server-core/src/main/java/com/yubico/webauthn/AssertionRequest.java:130]: I would remove the comma before "and"

File: webauthn-server-core/src/main/java/com/yubico/webauthn/AssertionRequest.java [hunk 6]
```diff
@@ -215,8 +223,10 @@ public AssertionRequestBuilder publicKeyCredentialRequestOptions(
      *
      * <p>If this is empty, this indicates that this is a request for an assertion by a <a
      * href="https://www.w3.org/TR/2021/REC-webauthn-2-20210408/#client-side-discoverable-public-key-credential-source">client-side-discoverable
-     * credential</a>, and identification of the user has been deferred until the response is
-     * received.
+     * credential</a> (passkey), and identification of the user has been deferred until the response
```
Comment (aveenismail) [inline webauthn-server-core/src/main/java/com/yubico/webauthn/AssertionRequest.java:226]: I would remove the comma before "and"

File: webauthn-server-core/src/main/java/com/yubico/webauthn/AssertionRequest.java [hunk 7]
```diff
@@ -230,8 +240,10 @@ public AssertionRequestBuilder username(@NonNull Optional<String> username) {
...
```
Comment (aveenismail) [inline webauthn-server-core/src/main/java/com/yubico/webauthn/AssertionRequest.java:243]: I would remove the comma before "and"

File: webauthn-server-core/src/main/java/com/yubico/webauthn/AssertionRequest.java [hunk 8]
```diff
@@ -250,8 +262,10 @@ public AssertionRequestBuilder username(String username) {
      * <p>If both this and {@link #username(String)} are empty, this indicates that this is a
      * request for an assertion by a <a
      * href="https://www.w3.org/TR/2021/REC-webauthn-2-20210408/#client-side-discoverable-public-key-credential-source">client-side-discoverable
-     * credential</a>, and identification of the user has been deferred until the response is
-     * received.
+     * credential</a> (passkey), and identification of the user has been deferred until the response
```
Comment (aveenismail) [inline webauthn-server-core/src/main/java/com/yubico/webauthn/AssertionRequest.java:265]: I would remove the comma before "and"

File: webauthn-server-core/src/main/java/com/yubico/webauthn/AssertionRequest.java [hunk 9]
```diff
@@ -266,8 +280,10 @@ public AssertionRequestBuilder userHandle(@NonNull Optional<ByteArray> userHandl
...
```
Comment (aveenismail) [inline webauthn-server-core/src/main/java/com/yubico/webauthn/AssertionRequest.java:283]: I would remove the comma before "and"

File: README [hunk 10]
```diff
@@ -367,31 +370,56 @@ throw new RuntimeException("Authentication failed");
 
 Finally, if the previous step was successful, update your database using the
 link:https://developers.yubico.com/java-webauthn-server/JavaDoc/webauthn-server-core/2.4.1/com/yubico/webauthn/AssertionResult.html[`AssertionResult`].
-Most importantly, you should update the signature counter. That might look something like this:
+Most importantly you should update the signature counter, and the backup state flag if you use it. That might look something like this:
```
[the remark this answers is not in the document]
Comment (aveenismail) [reply README:373]: I actually understood it as "if you use it" is referring to only the backup state even without the comman, but yeah, someone else might equate it with the signature counter..

File: webauthn-server-core/src/main/java/com/yubico/webauthn/AssertionRequest.java (hunk 2 above)
Comment (emlun) [reply webauthn-server-core/src/main/java/com/yubico/webauthn/AssertionRequest.java]: I agree this comma is awkward, but I don't like removing it either. I replaced it with a period and rewrote the subordinate clause a bit instead. :slightly_smiling_face:

File: README (hunk 1 above)
Comment (emlun) [reply README:411]: Thanks, but I feel like it flows better the way it is.

**Pull request merged**